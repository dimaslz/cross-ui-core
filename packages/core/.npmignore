.DS_Store
.git
.npmrc