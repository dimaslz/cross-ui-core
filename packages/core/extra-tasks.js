const fs = require('fs');
const path = require('path');
const shelljs = require('shelljs');

const nodeModulesDir = path.resolve(`${__dirname}/node_modules`);

function renameCssVarPrefix(css, prefix) {
  return css.replace(/--tw-/g, `--${prefix}-`);
}

function cleanCSS(css) {
  return css.replace('@tailwind utilities; ', '');
}

function moveSCSS() {
  shelljs.mkdir('./dist/scss');
  shelljs.mkdir('./dist/scss/components');

  shelljs.cp('./src/assets/css/variables.scss', './dist/scss/variables.scss');
  shelljs.cp(
    './src/components/button/button.scss',
    './dist/scss/components/button.scss',
  );
  shelljs.cp(
    './src/components/input-text/input-text.scss',
    './dist/scss/components/input-text.scss',
  );
  fs.writeFileSync(
    path.resolve(`${__dirname}/dist/scss/components/components.scss`),
    `@import '~tailwindcss/utilities.css';
@import 'button.scss';
@import 'intput-text.scss';`,
  );
}

function getVariables(css) {
  const originalUtilitiesCSS = fs.readFileSync(
    path.resolve(`${nodeModulesDir}/tailwindcss/dist/utilities.css`),
    'utf8',
  );

  let match = originalUtilitiesCSS.match(
    new RegExp('\\*\\s*\\{[^\\}]+\\}', 'gm'),
  );
  if (match) {
    let variables = match.join('\n');
    variables = variables
      .replace(/\n/gm, '')
      .replace(/\s+--/gm, '--')
      .replace(/\s+\{/gm, '{')
      .replace(/\s+/gm, ' ');

    return `${variables} ${css}`;
  }

  return css;
}

async function run() {
  console.log('üë®‚Äçüíª Making some last fixes...');
  const cssOrigin = fs.readFileSync(
    path.resolve(`${__dirname}/dist/cross-ui/cross-ui.css`),
    'utf8',
  );

  let cssUpdated = cssOrigin;
  cssUpdated = getVariables(cssUpdated);
  cssUpdated = renameCssVarPrefix(cssUpdated, 'cui');
  cssUpdated = cleanCSS(cssUpdated);

  fs.writeFileSync(
    path.resolve(`${__dirname}/dist/cross-ui/cross-ui.css`),
    cssUpdated,
  );

  moveSCSS();
}

run();
