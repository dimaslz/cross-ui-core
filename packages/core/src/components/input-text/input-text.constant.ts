export const enum TYPE {
  TEXT = 'text',
  PASSWORD = 'password',
  TELEPHONE = 'tel',
}
