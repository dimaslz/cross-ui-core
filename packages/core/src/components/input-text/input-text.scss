$prefix: 'cross_ui__';

.CrossUIInputText {
  @apply text-gray-600 bg-transparent block;
  label {
    @apply flex w-full;
  }

  @each $color in $colors {
    &.#{$prefix}#{$color} {
      $color-value: map-get($colors-map, $color);

      input,
      textarea {
        @apply w-full flex border outline-none rounded-sm;

        &:focus {
          @apply border-opacity-50 border-#{$color-value}-500;
        }
      }

      &.#{$prefix}small {
        @apply text-xs;
        input,
        textarea {
          @apply px-2 py-1;
        }
      }

      &.#{$prefix}medium {
        @apply text-sm;
        input,
        textarea {
          @apply px-2 py-2;
        }
      }

      &.#{$prefix}large {
        @apply text-sm;
        input,
        textarea {
          @apply px-2 py-3;
        }
      }
    }
  }

  &__hint,
  &__error {
    @apply text-xs py-1;
  }

  &__error {
    @apply text-red-400;
  }

  &__hint {
    @apply text-gray-400;
  }

  input:disabled {
    @apply opacity-50 cursor-not-allowed text-opacity-50 bg-gray-100;
  }

  input:placeholder,
  textarea:placeholder {
    @apply placeholder-gray-300;
  }

  &.#{$prefix}full-width {
    @apply w-full;
  }
}
