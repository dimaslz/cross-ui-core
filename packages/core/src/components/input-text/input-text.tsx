import {
  Component,
  Host,
  h,
  Prop,
  Listen,
  Element,
  State,
} from '@stencil/core';

const componentClass = 'CrossUIInputText';

enum TYPE {
  text = 'text',
  password = 'password',
  tel = 'tel',
}

@Component({
  tag: 'cross-input-text',
})
export class InputText {
  @Element() el: HTMLElement;
  @Prop({ mutable: true }) disabled: boolean = false;
  @Prop({ mutable: true }) required: boolean = false;
  @Prop({ mutable: true }) label: string = '';
  @Prop({ mutable: true }) type: string = TYPE.text;
  @Prop({ mutable: true }) value: string = '';
  @Prop({ mutable: true }) hint: string = '';
  @Prop({ mutable: true }) placeholder: string = '';
  @Prop({ mutable: true }) size: string = 'small';
  @Prop() color: string = 'primary';

  @Prop() multiline: boolean = false;
  @Prop() cols: number = null;
  @Prop() rows: number = null;

  @State() highlightRequired: boolean = false;
  @State() error: boolean = false;
  @State() errorMessage: string;

  componentWillLoad() {
    if (!Boolean(this.label && this.placeholder)) {
      console.warn(
        "[CROSS]: If you don't use label, placeholder attribute is mandatory",
      );
    }
  }

  @Listen('focus', { capture: true })
  onFocusHandler($event) {
    if (this.disabled) {
      $event.stopPropagation();
      return false;
    }

    (this.el as any)?.onBlur?.call(this, $event);
  }

  @Listen('paste', { capture: true })
  onPasteHandler($event) {
    if (this.disabled) {
      $event.stopPropagation();
      return false;
    }

    (this.el as any)?.onPaste?.call(this, $event);
  }

  @Listen('blur', { capture: true })
  onBlurHandler($event) {
    if (this.disabled) {
      $event.stopPropagation();
      return false;
    }

    this.error = !!(!Boolean(this.value) && this.required);
    if (this.error) {
      this.errorMessage = 'This field is required';
    }

    (this.el as any)?.onBlur?.call(this, $event);
  }

  @Listen('input', { capture: true })
  onInputHandler($event) {
    if (this.disabled) {
      $event.stopPropagation();
      return false;
    }

    this.value = $event.target.value;

    this.error = !!(!Boolean(this.value) && this.required);
    if (this.error) {
      this.errorMessage = 'This field is required';
    }

    (this.el as any)?.onChange?.call(this, $event);
    (this.el as any)?.onInput?.call(this, $event);
  }

  render() {
    const style: string = [
      componentClass,
      this.size,
      this.disabled,
      this.color,
    ].join(' ');

    return (
      <Host class={`${style}`}>
        {this.label && <label htmlFor="input">{this.label}</label>}
        {this.multiline ? (
          <textarea
            value={this.value}
            disabled={this.disabled}
            placeholder={this.placeholder}
            class={this.error ? 'border-red-300' : ''}
            id="input"
            cols={this.cols}
            rows={this.rows}
          ></textarea>
        ) : (
          <input
            type={this.type}
            value={this.value}
            disabled={this.disabled}
            placeholder={this.placeholder}
            class={this.error ? 'border-red-300' : ''}
            id="input"
          />
        )}
        {this.error && (
          <div class="CrossUIInputText__error">{this.errorMessage}</div>
        )}
        {this.hint && <div class="CrossUIInputText__hint">{this.hint}</div>}
      </Host>
    );
  }
}
