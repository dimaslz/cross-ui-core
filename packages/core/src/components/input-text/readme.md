# cross-input-text



<!-- Auto Generated Below -->


## Properties

| Property      | Attribute     | Description | Type      | Default     |
| ------------- | ------------- | ----------- | --------- | ----------- |
| `color`       | `color`       |             | `string`  | `'primary'` |
| `cols`        | `cols`        |             | `number`  | `null`      |
| `disabled`    | `disabled`    |             | `boolean` | `false`     |
| `hint`        | `hint`        |             | `string`  | `''`        |
| `id`          | `id`          |             | `string`  | `''`        |
| `label`       | `label`       |             | `string`  | `''`        |
| `multiline`   | `multiline`   |             | `boolean` | `false`     |
| `placeholder` | `placeholder` |             | `string`  | `''`        |
| `required`    | `required`    |             | `boolean` | `false`     |
| `rows`        | `rows`        |             | `number`  | `null`      |
| `size`        | `size`        |             | `string`  | `'small'`   |
| `type`        | `type`        |             | `string`  | `TYPE.text` |
| `value`       | `value`       |             | `string`  | `''`        |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
