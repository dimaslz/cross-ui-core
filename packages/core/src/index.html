<!DOCTYPE html>
<html dir="ltr" lang="en">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0, minimum-scale=1.0, maximum-scale=5.0" />
    <title>Stencil Component Starter</title>

    <script type="module" src="/build/cross-ui-core.esm.js"></script>
    <script nomodule src="/build/cross-ui-core.js"></script>
  </head>
  <body>
    <cross-playground></cross-playground>
  </body>
</html>
