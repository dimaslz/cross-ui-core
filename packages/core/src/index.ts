export { Components, JSX } from './components';

export { SIZES, SIZE, COLORS, COLOR } from './constants'