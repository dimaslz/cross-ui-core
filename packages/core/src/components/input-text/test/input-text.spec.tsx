import { newSpecPage } from '@stencil/core/testing';
import { InputText } from '../input-text';

describe('input-text', () => {
  it('renders', async () => {
    const page = await newSpecPage({
      components: [InputText],
      html: `<input-text></input-text>`,
    });
    expect(page.root).toMatchSnapshot();
  });
});
