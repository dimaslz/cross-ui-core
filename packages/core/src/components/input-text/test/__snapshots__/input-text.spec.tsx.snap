// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`input-text renders 1`] = `<input-text />`;
